infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -266,6 +271,58 @@ protected void internalStart(Map<String, String> startOptions) throws Infrastruc
     }
   }
 
+  private void fixCommands(KubernetesRuntimeContext<E> context) {
```

Comment:
To be seems this method must be rewritten to make logic clearer. Let me know if you would like to get some suggestions.

Reply:
Like you can consider reusing KubernetesServerResolver to resolve server url by port.

Reply:
reworked